src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,27 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  def takeRight(n: Int): Iterator[A] = {
+    val buffer = new Array[Any](n)
```

Comment:
`takeRight(Int.MaxValue)` should be a no-op, not consume all available memory.  For less extreme things, it's okay to have some overhead if you can't know the length, but eagerly allocating a huge chunk is not so good.  There should be a reasonable default size and then it should grow until it reaches the full requested size.